test/auth.js

```diff
@@ -135,6 +139,40 @@ describe('auth interface', () => {
   }).timeout(20000);
 });
 
+describe('Get granted containers permissions from auth URI', () => {
+  it('invalid uri', async () => {
+    const appNoConnect = createTestApp();
+    should(appNoConnect.auth.readGrantedPermissions('safe-invalid-uri')).be.rejectedWith('Serialisation error');
+  });
+
+  it('uri with no auth granted information', async () => {
```

Comment:
@bochaco this test... should it be rejected?

I mean... it's succeeding, there's just no perms granted to be shown, right?

Reply:
I think I understand what you mean, I was thinking that an auth URI like this one which is for unregistered connection (no containers permissions information in it even that it can be decoded) could be triggering an error to make the difference with a URI which does contain containers permissions (regardless the number of permissions). E.g. you can have an auth URI where you didn't request permissions for any container, therefore the returned object will be empty, I just added a test for this case and it expects to succeed and return an empty object.

Also, the other two types of auth URI's, i.e. for share MD and containers permissions, don't contain any information, they just have a request id and the containers permissions granted are just updated on the network, so for those cases the app will need to use the function `getContainersPermissions` after it's connected.

Reply:
Aha, yeh, that's what I was meaning.

If a non-authed app uri can never have permissions / would never be correct, then I'd say we change the error message to reflect this.

So we can have 1) authed app uri with no perms: return empty object.
2) unauthed app uri, which cannot have perms: throw error detailing that fact.

(maybe `This uri is not for an authenticated app` or something? so you can see your mistake)